frontend/src/app/commonComponents/MultiSelect/MultiSelect.test.tsx

```diff
@@ -17,85 +17,130 @@ const fruitOptions: MultiSelectDropdownOption[] = [
+  describe("selecting and unselecting", () => {
+    beforeEach(() => {
+      renderWithoutInitialOptions();
+    });
+    describe("when selecting an item", () => {
       beforeEach(() => {
-        const pillContainer = screen.getByTestId("pill-container");
-        expect(pillContainer).toBeInTheDocument();
-        userEvent.click(
-          within(pillContainer).getByTestId("Apples-pill-delete")
-        );
         userEvent.click(screen.getByTestId("multi-select-input"));
+        const optionList = screen.getByTestId("multi-select-option-list");
+        userEvent.click(within(optionList).getByText("Apples"));
+        userEvent.click(within(optionList).getByText("Oranges"));
       });
 
-      it("should remove the pill ", () => {
+      it("should show the pill with the selected item's label", () => {
         const pillContainer = screen.getByTestId("pill-container");
         expect(pillContainer).toBeInTheDocument();
 
-        expect(pillContainer.children.length).toEqual(1);
+        // children includes a legend, so length is 1 more than selected items
+        expect(pillContainer.children.length).toEqual(3);
+        within(pillContainer).getByText("Apples");
         within(pillContainer).getByText("Oranges");
       });
-      it("should make the option available to select", () => {
+      it("should show the remove the selected options from the options list", () => {
         const optionList = screen.getByTestId("multi-select-option-list");
         expect(optionList).toBeInTheDocument();
-        expect(optionList.children.length).toEqual(5);
+        expect(optionList.children.length).toEqual(4);
+        expect(
+          within(optionList).queryByText("Apples")
+        ).not.toBeInTheDocument();
         expect(
           within(optionList).queryByText("Oranges")
         ).not.toBeInTheDocument();
       });
+
+      describe("when deleting a selected item", () => {
+        beforeEach(() => {
+          const pillContainer = screen.getByTestId("pill-container");
+          expect(pillContainer).toBeInTheDocument();
+          userEvent.click(within(pillContainer).getAllByRole("button")[0]);
+          userEvent.click(screen.getByTestId("multi-select-input"));
+        });
+
+        it("should remove the pill ", () => {
+          const pillContainer = screen.getByTestId("pill-container");
+          expect(pillContainer).toBeInTheDocument();
+
+          // children includes a legend, so length is 1 more than selected items
+          expect(pillContainer.children.length).toEqual(2);
+          within(pillContainer).getByText("Oranges");
+        });
+        it("should make the option available to select", () => {
+          const optionList = screen.getByTestId("multi-select-option-list");
+          expect(optionList).toBeInTheDocument();
+          expect(optionList.children.length).toEqual(5);
+          expect(
+            within(optionList).queryByText("Oranges")
+          ).not.toBeInTheDocument();
```

Comment:
nit: I think this test is asserting the opposite of what we expect?

Reply:
@emmastephenson I took a look back through the whole test suite, and this test case is correct. It's a bit hard to follow because the suite was written with multiple nested `beforeEach` statements, which sets things up such that for this last test case `Oranges` actually isn't in the list. Personally I'd like to see us [move away from so much nesting in testing](https://kentcdodds.com/blog/avoid-nesting-when-youre-testing) b/c it makes the tests really hard to read/refactor and too inder-dependent, giving us small improvements in efficiency and a big loss of readability. For the sake of not sending this PR through another review though I'm gonna leave this as is. 🙃